src/mqt/ionshuttler/multi_shuttler/gate_partitioning_tabu.py

```
@@ -0,0 +1,1535 @@
+    qubits: Sequence[int],
+    required_edges: dict[tuple[int, int], float],
+) -> tuple[list[_Supernode], dict[int, int]]:
+    """Merge qubits that must travel together within the slice.
+
+    Any qubits connected through the required two-qubit interactions are folded
+    into the same supernode. The optimizer then automatically moves that supernode
+    as a single unit instead of trying to keep the member qubits synchronized manually.
+    """
+
+    union_find = _UnionFind(list(qubits))
+    for (left, right), _weight in required_edges.items():
```

Ruff (0.15.12)

[warning] When using only the keys of a dict use the `keys()` method

Replace `.items()` with `.keys()`

(PERF102)